src/pal/src/locale/unicode.cpp

```diff
@@ -438,16 +295,13 @@ GetCPInfo(
         goto done;
     }
 
-    if ( NULL != ( lpStruct = CODEPAGEGetData( CodePage ) ) )
-    {
-        lpCPInfo->MaxCharSize = lpStruct->nMaxByteSize;;
-        memcpy( lpCPInfo->LeadByte, lpStruct->LeadByte , MAX_LEADBYTES );
+    lpCPInfo->MaxCharSize = 4;
+    memset( lpCPInfo->LeadByte, 0, MAX_LEADBYTES );
```

Comment:
@jkotas, since this zero-inits the LeadByte buffer and does not fill the array, does this mean code after this line is dead: https://github.com/dotnet/runtime/blob/9dc349f8685629595e7bb026dddb549017e9dbb2/src/coreclr/src/pal/src/locale/unicode.cpp#L291

I was chasing down the remaining `UNIXTODO` in runtime repo when I found `IsDBCSLeadByte` and then this. If we don't care about any LeadByte-related functionality on Unix, maybe that TODO can be removed from PAL? Otherwise, I have a `IsDBCSLeadByte` implementation for `CP_ACP` with a constant array containing some values and zero holes, can send a PR to address the `UNIXTODO`. 🙂

Reply:
We have Unix hard-coded to UTF8.  `IsDBCSLeadByte` / `IsDBCSLeadByteEx` functions are broken for UTF8. UTF8 can have 1, 2 or 3 bytes per character. DBCS encoding assume that one character is 1 or 2 bytes. 

If these functions are called and the input is 3 byte unicode character, the result is going to be incorrect.

It would be best to delete `IsDBCSLeadByte`, `IsDBCSLeadByteEx` and all other methods related to DBCS encodings, and fix the code that calls them accordingly.

Reply:
Opened https://github.com/dotnet/runtime/pull/41611 for cleanups.